### PR TITLE
[AEA-950] Add libp2p bench

### PR DESCRIPTION
**Pull request by lrahmani**
## Proposed changes

Add benchmark files used in the paper

### CODE REVIEW

File: packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/dhtpeer.go [hunk 1]
```diff
@@ -831,7 +837,16 @@ func (dhtPeer *DHTPeer) registerAgentAddress(addr string) error {
 	linfo().Str("op", "register").
 		Str("addr", addr).
 		Msgf("Announcing address to the dht with cid key %s", addressCID.String())
+	//var elapsed time.Duration
+	//start := time.Now()
 	err = dhtPeer.dht.Provide(ctx, addressCID, true)
+	//elapsed = time.Since(start)
```
Comment (panasevychol) [inline packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/dhtpeer.go:843]: Do we need this commented code here and above? If yes maybe it worth to have a comment that explains why it's here.

File: packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/benchmarks_test.go [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+/* **********
+* How to run
+* ***********
+
+	$ go test -p 1 -count 20 libp2p_node/dht/dhtpeer/ -run=XXX  -bench .  -benchtime=20x
+
+*/
+
+var keysFilePath = "/path/to/file/benchmark_keys.txt"
```
Comment (solarw) [inline packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/benchmarks_test.go:43]: this path a bit confusing, probably I miss something?

**Review by solarw — APPROVED** [review]
LGTM

File: packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/benchmarks_test.go [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+var addrsFilePath = "/path/to/file/benchmark_addrs.txt"
+var tcpUri = "localhost:12345"
+
+/* **********************************
+ * baseline TCP connection benchmark
+ * ********************************** */
+
+// helpers
+
+func acceptAndEcho(server net.Listener) {
```
Comment (panasevychol) [inline packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/benchmarks_test.go:53]: Just wondering with the level of nesting in this function. Wasn't it possible to move from level 3 to a separate func?

**Review by panasevychol — APPROVED** [review]
I didn't dig much but brief impression is good. Just a few comments rather then changes requests.
And pls fix the conflicts with a target branch.

File: packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/benchmarks_test.go (hunk 3 above)
Comment (lrahmani) [reply packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/benchmarks_test.go:59]: you are right, considering that it is intended to be used only once I didn't want to add an extra symbol for it

File: packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/benchmarks_test.go (hunk 2 above)
Comment (lrahmani) [reply packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/benchmarks_test.go]: addressed

File: packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/dhtpeer.go (hunk 1 above)
Comment (lrahmani) [reply packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/dhtpeer.go]: addressed

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**